manifests/common/neutron.pp

```diff
@@ -32,20 +34,22 @@
     region           => $::openstack::config::region,
   }
 
-  class { '::neutron::server':
-    auth_host           => $::openstack::config::controller_address_management,
-    auth_password       => $::openstack::config::neutron_password,
-    database_connection => $::openstack::resources::connectors::neutron,
-    enabled             => $::openstack::profile::base::is_controller,
-    sync_db             => $::openstack::profile::base::is_controller,
-    mysql_module        => '2.2',
-  }
+  if $is_controller {
+    class { '::neutron::server':
+      auth_host           => $::openstack::config::controller_address_management,
+      auth_password       => $::openstack::config::neutron_password,
+      database_connection => $::openstack::resources::connectors::neutron,
+      enabled             => $is_controller,
+      sync_db             => $is_controller,
+      mysql_module        => '2.2',
+    }
```

Comment:
The network node needs this class in order to set authentication credentials in /etc/neutron/neutron.conf. Could you make the `if $is_controller` just surround the neutron::server::notifications class, and then you can calso add `package_ensure => $is_controller` here? As in

```
class { '::neutron::server':
   auth_host           => $::openstack::config::controller_address_management,
   auth_password       => $::openstack::config::neutron_password,
   database_connection => $::openstack::resources::connectors::neutron,
   package_ensure      => $is_controller,
   enabled             => $is_controller,
   sync_db             => $is_controller,
   mysql_module        => '2.2',
 }

if $is_controller {
  class { '::neutron::server::notifications':
    # ...
```